### ISSUE TITLE
P1: ONNX import — and resolve the advertised-but-stubbed ONNX export

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#3152]
Parent: #3146 · Tier **P1 — breadth** · Milestone **0.69.0**

## The gap

ONNX is advertised and not implemented. `crates/apr-cli/src/commands/export.rs:7`:

```rust
//! - ONNX (.onnx) - Cross-framework inference (planned)
```

`ExportFormat::Onnx` exists in the enum and the error string offers `onnx` as a valid choice, so `apr export --format onnx` is reachable from the CLI while the implementation is a stub. Per the 0.63.0 dogfood finding class (advertised-but-absent surfaces), an advertised format that does not work is a defect in its own right.

More importantly, the direction that matters for adoption is the one that is entirely missing: **import**. Burn ships `burn-onnx`, which converts an ONNX graph into Rust code against Burn's native API. aprender has no path for "I have a model from somewhere else that is not GGUF or SafeTensors."

## Scope

1. **`apr import model.onnx`** — parse the ONNX protobuf, map a documented operator subset onto aprender/realizar ops, emit a runnable `.apr`.
2. Operator subset for v1 (matching the shape of what Burn supports, which is also a limited set): `Gemm`, `MatMul`, `Conv`, `Relu`, `Sigmoid`, `Tanh`, `MaxPool`, `AveragePool`, `GlobalAveragePool`, `BatchNormalization`, `Softmax`, `Add`, `Mul`, `Reshape`, `Transpose`, `Concat`, `Flatten`, `Gather`.
3. **Decide and record** what `apr export --format onnx` does: implement it, or remove `Onnx` from the reachable format list. Leaving it reachable-and-stubbed is not an option.

## Acceptance criteria (each must be able to FAIL)

- [ ] **Unsupported operators fail loudly, naming the op.** A fail-closed test feeds a model containing an op outside the subset and asserts the error message contains that op's name. Silent partial import is the defect this criterion exists to exclude — per the repo rule, a blacklist that fails open on its complement is not a gate.
- [ ] Round-trip numerical parity: for each supported op, an ONNX model exercising it is imported and its output matches `onnxruntime` on the same input within `1e-5`. Fixtures pinned under `tests/fixtures/onnx/`.
- [ ] End-to-end: a real small ONNX model (e.g. a MobileNet-class classifier) imports and produces the same top-1 label as onnxruntime on 20 pinned inputs.
- [ ] Opset version is checked, and an unsupported opset is refused with the version in the message.
- [ ] The supported-operator list lives in a contract YAML, not only in a doc comment, so it cannot drift from the implementation.
- [ ] `apr import --help` and the command registry (`contracts/apr-cli-commands-v1.yaml`) agree, and `crates/apr-cli/tests/cli_commands.rs::registered_commands` is updated — FALSIFY-CLI-001/002 must stay green.
- [ ] If export is removed rather than implemented: the CLI no longer offers `onnx` in its format list, and a test asserts that.

**Comment by noahgift** [comment]
triage 2026-09-23 (aprender-dd, standing triage owner): milestoned to 0.74.0, theme Any Model (architectures, formats); it moved out of `backlog` (the 2026-09-17 overflow bucket, not a release).
